RIOT-OS/RIOT
# drivers/periph/rtt: enhancements to make RTT_FREQUENCY easily configurable

### Contribution description

This PR wants to make `RTT_FREQUENCY` easily configurable. Currently `RTT_FREQUENCY` is for most platforms a fixed value, I would like to be easily configurable at compile time.

I looked at most `rtt` implementations and in most cases `RTT_FREQUENCY` would be a a prescaled value of the clock source . Therefore I think a `min` and `max` value should be enough to map all possible values for `RTT_FREQUENCY`.

I added a `RTT_CLOCK_FREQUENCY` to make the clock source transparent, also useful for deriving prescaler values from the desired `RTT_FREQUENCY`.

I also force `RTT_MAX_VALUE` to be defined for all platforms.

I migrated all `stm32f1` to implement the required values. I'm willing to do the same for all other `rtt` configurations if the global change makes sense.

### Testing procedure

- `tests/periph_rtt` should still pass on `stm32f1` boards with different frequencies


`BOARD=iotlab-m3 make -C tests/periph_rtt flash test --no-print-directory -j3`
```
main(): This is RIOT! (Version: 2020.07-devel-56-g42700-pr_rtt_enhancement)

RIOT RTT low-level driver test
This test will display 'Hello' every 5 seconds

Initializing the RTT driver
RTT now: 0
Setting initial alarm to now + 5 s (81920)
Done setting up the RTT, wait for many Hellos
Hello
Hello
Hello
Hello
Hello
```

`CFLAGS+=-DRTT_FREQUENCY=1 BOARD=iotlab-m3 make -C tests/periph_rtt clean flash test --no-print-directory -j3`

```
main(): This is RIOT! (Version: 2020.07-devel-56-g42700-pr_rtt_enhancement)

RIOT RTT low-level driver test
This test will display 'Hello' every 5 seconds

Initializing the RTT driver
RTT now: 0
Setting initial alarm to now + 5 s (5)
Done setting up the RTT, wait for many Hellos
Hello
Hello
Hello
Hello
Hello
```

- Invalid `RTT_FREQUENCY` are checked at compile time:

```
CFLAGS=-DRTT_FREQUENCY=0 BOARD=iotlab-m3 make -C tests/periph_rtt/ --no-print-directory
Building application "tests_periph_rtt" for "iotlab-m3" with MCU "stm32f1".

In file included from /home/francisco/workspace/RIOT/tests/periph_rtt/main.c:30:
/home/francisco/workspace/RIOT/drivers/include/periph/rtt.h:107:2: error: #warning "RTT_FREQUENCY is out of the allowed range, " "RTT_MIN_FREQUENCY < RTT_FREQUENCY < RTT_MAX_FREQUENCY" [-Werror=cpp]
 #warning "RTT_FREQUENCY is out of the allowed range, " \
  ^~~~~~~
cc1: all warnings being treated as errors
/home/francisco/workspace/RIOT/Makefile.base:110: recipe for target '/home/francisco/workspace/RIOT/tests/periph_rtt/bin/iotlab-m3/application_tests_periph_rtt/main.o' failed
make[1]: *** [/home/francisco/workspace/RIOT/tests/periph_rtt/bin/iotlab-m3/application_tests_periph_rtt/main.o] Error 1
/home/francisco/workspace/RIOT/tests/periph_rtt/../../Makefile.include:538: recipe for target '/home/francisco/workspace/RIOT/tests/periph_rtt/bin/iotlab-m3/application_tests_periph_rtt.a' failed
make: *** [/home/francisco/workspace/RIOT/tests/periph_rtt/bin/iotlab-m3/application_tests_periph_rtt.a] Error 2
```

### Issues/PRs references
